pkg/frontend/assets/flowhouse.js

```diff
@@ -284,64 +284,128 @@ function renderChart(rdata) {
   var chart = new google.visualization.AreaChart(document.getElementById('chart_div'));
   chart.draw(chartData, options);
 
-  const customLegendDiv = document.getElementById('custom_legend');
-  customLegendDiv.innerHTML = `
-  <div class="legend-help-tip">
-    <strong>How to use:</strong><br>
-    Click a flow to show only it. Click again to go back to all flows.<br>
-    Ctrl/Cmd/Option + click: add/remove flows.
-  </div>
-`;
-
-  const colors = options.colors;
-  const columns = data[0].length;
-
-  const table = document.createElement('table');
-  table.classList.add('table', 'table-sm', 'table-bordered');
-  const tbody = document.createElement('tbody');
-
-  for (let i = 1; i < columns; i++) {
-    const row = document.createElement('tr');
-    const colorCell = document.createElement('td');
-    colorCell.style.backgroundColor = colors[(i - 1) % colors.length];
-    colorCell.style.width = '20px';
-    const labelCell = document.createElement('td');
-    labelCell.textContent = data[0][i];
-    row.appendChild(colorCell);
-    row.appendChild(labelCell);
-    tbody.appendChild(row);
-
-    if (!window.seriesVisibility[i - 1]) {
-      row.style.opacity = '0.4';
-    } else {
-      row.style.opacity = '1.0';
+  renderLegendTable();
+
+  function renderLegendTable() {
+    const flowStats = [];
+    for (let i = 1; i < data[0].length; i++) {
+      let max = -Infinity;
+      for (let j = 1; j < data.length; j++) {
+        const val = data[j][i];
+        if (typeof val === "number" && !isNaN(val)) {
+          if (val > max) max = val;
+        }
+      }
+      flowStats.push({
+        index: i,
+        label: data[0][i],
+        max: max === -Infinity ? 0 : max
+      });
     }
 
-    row.addEventListener('click', function(event) {
-      const visibleCount = window.seriesVisibility.filter(Boolean).length;
-      if (event.ctrlKey || event.metaKey || event.altKey) {
-        // Ctrl/Cmd/Option + click: toggle this series
-        if (window.seriesVisibility[i - 1] && visibleCount === 1) {
-          // If this is the only visible series and user tries to hide it, show all
-          window.seriesVisibility = Array(columns - 1).fill(true);
+    // Sorting logic
+    if (!window.legendSort) window.legendSort = { key: "label", asc: true };
+    const sortKey = window.legendSort.key;
+    const sortAsc = window.legendSort.asc;
+
+    flowStats.sort((a, b) => {
+      if (sortKey === "label") {
+        return sortAsc ? a.label.localeCompare(b.label) : b.label.localeCompare(a.label);
+      } else {
+        return sortAsc ? a.max - b.max : b.max - a.max;
+      }
+    });
```

Comment:
[pebble] might be a little naive to assume only two sort options. it's correct now, but might not be in the future. maybe use a switch case?

Also you can simplify the `sortAsc` stuff.

```suggestion
    let sortModifier = 1;
    if (sortAsc) {
      sortModifier = -1;
    }
    flowStats.sort((a, b) => {
      switch (sortKey) {
        case "label":
          return a.label.localeCompare(b.label) * sortModifier;
        case "max":
          return (a.max - b.max) * sortModifier;
      }
      return 0;
    });
```

Reply:
changed 👍